### PR TITLE
Patch/patch is ticket winner

### PR DESCRIPTION
**Pull request by sternhenri**
slight refactor of https://github.com/filecoin-project/lotus/pull/808

Idea being not to assume that all sectors count all their storage toward power (who knows) but rather pull from the proving set (as `GetSectorsForElectionPost` does)

### CODE REVIEW

**Comment by whyrusleeping** [comment]
@sternhenri can you rebase this on testnet-staging?

File: chain/sync.go [hunk 1]
```diff
@@ -511,10 +511,13 @@ func (syncer *Syncer) ValidateBlock(ctx context.Context, b *types.FullBlock) err
 			return xerrors.Errorf("failed to get sector size for block miner: %w", err)
 		}
 
-		snum := types.BigDiv(mpow, types.NewInt(ssize))
+		mps, err := stmgr.GetMinerProvingSet(ctx, syncer.sm, baseTs, h.Miner)
```
Comment (magik6k) [inline chain/sync.go]: This isn't a super cheap call, and the math we had here previously is exactly the same.

Comment (sternhenri) [reply chain/sync.go]: It is not in a world in which you don't assume storage power across sectors is a multiple of `sectorSize` (if you get my gist....).

Lotus' call, but this is more future proof

**Comment by magik6k** [comment]
Lots of stuff changed around power after switch to specs-actors, if we need this it's probably easier to rewrite this from scratch

**Comment by sternhenri** [comment]
Noted. Lmk if I can help. Point remains which is you should probs not just assume sectors all contribute same power (esp now with the specs-actors changes)

**Pull request closed without merging**